File: src/easy/t2/solve.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        min_price = prices[0]
        for i in range(1, len(prices)):
            profit = max(profit, prices[i] - min_price)
            min_price = min(min_price, prices[i])
        return profit


solution = Solution()
prices = [7, 1, 5, 3, 6, 4]

print(solution.maxProfit(prices))

File: src/easy/t2/solve.go
package main

import "fmt"

func main() {
	var prices = []int{7, 1, 5, 3, 6, 4}
	fmt.Println(maxProfit(prices))

}

func maxProfit(prices []int) int {
	max := 0
	min := prices[0] // 最便宜的一天
	for i := 1; i < len(prices); i++ {
		if prices[i] < min {
			// 买入啊
			min = prices[i]
		}
		if dif := prices[i] - min; dif > max {
			max = dif
		}

	}
	return max
}
